### PR TITLE
fix: browse files instead of folders when adding DCC Tools on MacOS

### PR DESCRIPTION
**Pull request by seandillon92**
on macOS, when adding a DCC tool through the Preferences panel, the dialog used is a select folder dialog. Applications on macOS are distributed as .app files. As a result, the "Add DCC Tool" button is unusable for macOS users.

This PR fixes the issue by using a select file dialog when the OSX Editor is used.

### CODE REVIEW

File: Editor/Scripts/UserSettings/DCCToolsSettingsTab.cs [hunk 1]
```diff
@@ -124,20 +124,48 @@ internal class DCCToolsSettingsTab : IMeshSyncSettingsTab{
 //----------------------------------------------------------------------------------------------------------------------        
 
     #region Button callbacks
+
+    private const string addDCCToolLabel = "Add DCC Tool";
+    
+    string GetFilePath() {
```
Comment (sindharta) [inline Editor/Scripts/UserSettings/DCCToolsSettingsTab.cs]: Can you please change the name to indicate that this opens a FilePanel ?

File: Editor/Scripts/UserSettings/DCCToolsSettingsTab.cs [hunk 2]
```diff
@@ -124,20 +124,48 @@ internal class DCCToolsSettingsTab : IMeshSyncSettingsTab{
...
+        var path = EditorUtility.OpenFilePanel(addDCCToolLabel, m_lastOpenedFolder, "");
+        
+        if (!string.IsNullOrEmpty(path)) {
+            m_lastOpenedFolder = Path.GetDirectoryName(path);
+        }
+        
+        return path;
+    }
+
+    string GetFolderPath() {
```
Comment (sindharta) [inline Editor/Scripts/UserSettings/DCCToolsSettingsTab.cs]: Can you please change the name to indicate that this opens a FolderPanel ?

File: Editor/Scripts/UserSettings/DCCToolsSettingsTab.cs (hunk 1 above)
Comment (seandillon92) [reply Editor/Scripts/UserSettings/DCCToolsSettingsTab.cs]: done

File: Editor/Scripts/UserSettings/DCCToolsSettingsTab.cs (hunk 2 above)
Comment (seandillon92) [reply Editor/Scripts/UserSettings/DCCToolsSettingsTab.cs]: done

**Review by SonnyCampbellUnity — APPROVED** [review]

**Pull request merged**